### ISSUE TITLE
[REVIEW]: Probabilistic Biostatistics with Julia

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#36]
**Submitting author:** @tszanalytics (<a href="http://orcid.org/0000-0002-2469-1796">Jeffrey A Mills</a>)
**Repository:** <a href="https://github.com/tszanalytics/Juliacon2019" target ="_blank">https://github.com/tszanalytics/Juliacon2019</a>
**Editor:** @matbesancon
**Reviewer:** @bkamins, @kescobo
**Archive:** Pending

## Status

[![status](https://proceedings.juliacon.org/papers/a0a30e56a0e272f19828cea239270c93/status.svg)](https://proceedings.juliacon.org/papers/a0a30e56a0e272f19828cea239270c93)

Status badge code:

```
HTML: <a href="https://submissions.juliacon.org/papers/a0a30e56a0e272f19828cea239270c93"><img src="https://submissions.juliacon.org/papers/a0a30e56a0e272f19828cea239270c93/status.svg"></a>
Markdown: [![status](https://submissions.juliacon.org/papers/a0a30e56a0e272f19828cea239270c93/status.svg)](https://submissions.juliacon.org/papers/a0a30e56a0e272f19828cea239270c93)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/tszanalytics/Juliacon2019" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@bkamins & @kescobo, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/JuliaCon/proceedings-reviews/invitations

The reviewer guidelines are available here: https://proceedings.juliacon.org/guide/reviewers. Any questions/concerns please let @matbesancon know.

✨ **Please try and complete your review in the next two weeks** ✨

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/tszanalytics/Juliacon2019">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@tszanalytics) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
~- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)~

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
~- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support~

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/tszanalytics/Juliacon2019">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@tszanalytics) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
~~**Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support~~

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @bkamins, @kescobo it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/JuliaCon/proceedings-review) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/JuliaCon/proceedings-review:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00036/jcon.00036/10.21105.jcon.00036.pdf)

**Comment by tszanalytics** [comment]
Just an update:  I am putting together example code to generate all the figures and the table in the paper.  I will post this on the repo soon.  I've add a license.  I have added a table and made some other changes to the paper, and I am still working on editing the paper - I will upload a revised version soon.  I will also add the conference presentation and any related code to the repo once I have that together.

**Comment by bkamins** [comment]
Thank you for all this. So probably it is best if we wait for the update before reviewing. Right?
Please let us know when you are done. Thank you.

Some minor points while you update the text are (this is based on just skimming over the paper - I will read it in derail after the final version is uploaded):
* could you please increase the font side a bit in Fig. 1, 3, 4, 5 (to match the font size of the text) - if possible (BTW: Fig. 2 is a work of art 👍)
* it would be best to avoid breaking the code between pages (like now between page 3 and 4)
* in the code that "overflows" page end indentations are missing in `begin`-`end` and `for` code blocks
* another minor issue (in all codes) is to use spaces consistently (between operators, after commas etc.)
* Citations in IEEE style are usually given before a comma, see http://ieeeauthorcenter.ieee.org/wp-content/uploads/IEEE_Style_Manual.pdf page 12, for a reference
* It would be great if you added a one paragraph on timings of the computations (you only say that it is expensive computationally at the end of the paper now) - sorry if I missed this and you already do it (I just only glanced over the paper now)

**Comment by tszanalytics** [comment]
Yes, waiting for the update before reviewing makes sense.  I will incorporate all your suggestions above, thanks for these!

**Comment by kescobo** [comment]
Alright - I'm back. Looks like a lot has happened, but we're waiting on some updates before moving forward with review. Is that right?

**Comment by tszanalytics** [comment]
Yep, working on my presentation right now!  Will update this after the presentation.

**Comment by bkamins** [comment]
Some small code review comments:
* n your text `Array{Real}` can be probably replaced by `Array{Float64}` and the code shall run faster.
* you define `wt_draws` twice but do not define `wp_draws` below in the code and probably you do not need to call `Array` when you extract values from `ct`

(I can check it if we have reproducible codes)

**Comment by tszanalytics** [comment]
Thanks Bogumil!  I had spotted the wt_draws error but not the Real.  Also, thanks for answering Stefan K's question more clearly than I did at my presentation - I chatted with him later and he said you had explained it to him.

At the airport in Baltimore waiting for my flight right now.  Wish I could have stayed for the hackathon.  Was a fantastic conference.  Hope to chat with you at JuliaCon 2020!

**Comment by bkamins** [comment]
@tszanalytics - thank you for a very nice presentation (and I did not think that you knew that it was me who made the remark - but it turns out you did 😄).

I am obsessed with the difference in interpretation of hypothesis test results in frequentist and Bayesian inference (and actually I think that Bayesian interpretation is more intuitive for most people) and your work fits perfectly into this picture.

**Comment by tszanalytics** [comment]
@bkamins - Actually I didn't know it was you at the time, I only found out when I chatted with Stefan later.  I wish I had known so I could have chatted with you.  We must get together at the next time.  My flight delayed so more time to work on the proceedings revisions 😄

**Comment by tszanalytics** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00036/jcon.00036/10.21105.jcon.00036.pdf)

**Comment by tszanalytics** [comment]
@bkamins   Ok, recompiled the pdf here.  Thanks for pointing me in the right direction.

**Comment by bkamins** [comment]
@tszanalytics 

I think your work is an interesting case study. Here are my comments to your submission (please treat them as suggestions and decide what you think should be incorporated).

### Formal requirements of the review process (boxes checked-off above)
* **Installation instructions:**: please give information about Julia and package versions used in your work or add Project.toml that specifies this
* **Community guidelines:**: please add information whom to contact in case there are bugs/problems in your code
* **Page limit:**: last page has only a few lines; usually it is recommended to avoid such layout in publications; you can consider rectifying it (while keeping in line with the page limit specified by the proceedings guidelines)

### Code related
* `rng_example.jl`:
    * it is typically recommended to add a space after a comma in the code (also in other files)
    * missing `using Random` on top of the file
    * missing `using StatsPlots` on top of the file
    * probably last line `# savefig("approaching_certainty_03.png")` should be removed from the file as it is commented out
* `logistic_trajectory_by_week_example.jl`
    * `https://github.com/tszanalytics/BayesTesting.jl` is not a registered package; can you please add instructions how it should be loaded by the user (in particular which commit should be used as a reference); also please place all `using` statements at the top of the file (now they are in several places and there are duplicates)
    * there are commented-out lines of code in the file - are they needed?
    * in line 121 you can consider fixing the indentation of the code
* `juliacon_proceedings2019_code.jl`
    * similar comments as for `logistic_trajectory_by_week_example.jl`
    * additionally you can consider breaking very long lines into several lines, like line 132

### Paper related

A major comment is to give more discussion of your BayesTesting.jl package (design goals, functionality etc. - see questions for reviewers to see what kind of information is expected). Of course please try to keep in the page limit of the proceedings when doing this (this is a soft suggestion).

**Comment by bkamins** [comment]
+ some comments to the PDF in the attached file.
[10.21105.jcon.00036.pdf](https://github.com/JuliaCon/proceedings-review/files/3490508/10.21105.jcon.00036.pdf)

**Comment by matbesancon** [comment]
@tszanalytics did you finish the corrections from @bkamins ?

**Comment by kescobo** [comment]
I'm a bit confused about whether I'm supposed to be evaluating just the code in the Proceedings repo, `BayesTesting.jl` or both. This review assumes both, please let me know if I'm mistaken.

## Overall

This manuscript presents a justification for and examples of using Bayesian inference in biomedical research, specifically outlining a method using julia to generate "Bayesian p-values" as a bridge from the current frequentist methods typically employed in biological contexts. As a biologist that's just starting to learn about Bayesian methods, I'm excited to have this resource and to have a force in the julia community pushing towards more robust statistical methods.

The manuscript could benefit from some minor updates to present a consistent voice and to provide additional explanation / context in the form of citations for a number of claims. The accompanying code should have substantially more documentation / clarity, both in terms of content and presentation. That said, this effort is definitely worth publishing with some modification. 

### Manuscript

Comments on the manuscript can be found here: 
[10.21105.jcon.00036_KSB_comments.pdf](https://github.com/JuliaCon/proceedings-review/files/3522868/10.21105.jcon.00036_KSB_comments.pdf)

### Code

I largely agree with @bkamins ' comments here regarding the need for additional documentation (in a perfect world, there would be runable jupyter notebooks, or you could use something like `Literate.jl`) and tests.

### Other Minor Comments

- There's a "Third Author" in the authors list. I'm guessing that's part of the template slipping through?
- https://github.com/tszanalytics/BayesTesting.jl/issues/11
- Is the raw data for the analyses you perform in the paper accessible? Would be great to have a `DataDeps` annotation for it so it's easily acquirable to reproduce your analyses.

**Comment by matbesancon** [comment]
Indeed that's both, thanks

**Comment by tszanalytics** [comment]
Thanks for the comments all.  I have been a bit busy the last week or so, but now this is top of my list.  I will get to work revising the paper and the code, etc. to address the comments.  Adding a jupyter notebook sounds like a good idea also, so I will have a go at putting one together.

**Comment by matbesancon** [comment]
ping @tszanalytics, did you have time to work on the corrections?

**Comment by tszanalytics** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00036/jcon.00036/10.21105.jcon.00036.pdf)

**Comment by tszanalytics** [comment]
@matbesancon 
@bkamins 
@kescobo 

Thanks everyone for all the great comments and suggestions.  I think the paper is greatly improved as a result.  I believe everything on the github repo is now up to date and all the changes are incorporated there.  Here's a summary of many of the revisions in response to the reviewer suggestions:

•	Changed title.
•	Added email addresses.
•	Put citations before periods and commas.
•	Fixed typos in equations (3) and (4) and added cite to Gelman (2004).
•	Tried changing from Real to Float64, but for this model it turns out it has to be Real because of 
use of auto-differentation.
•	Reworded 1st sentence in intro. to make it clearer.
•	Reworded 3rd paragraph so it makes more sense and to remove “clean code”.
•	Provided a citation to support the “iron law” and “institutional inertia” statements.
•	Provided a citation for the Behrens-Fisher problem.
•	Edited references section to correct several typos, etc.
•	Fixed typos in the figures.
•	Increased the font size for the figures to the extent possible (I found this to be particularly difficult!).
•	Reduced the size of Fig. 3 by only including two sets of time plot + density plot, and explained its purpose further. This allowed the addition of a new figure (Fig. 5) that provides an example of posterior density plots for treatment, placebo and difference.
•	Reduced the length of the paper so that it fits on 5 pages.

Please let me know if I've missed anything and I will endeavor to fix it.

Thanks,
Jeff

**Comment by kescobo** [comment]
Thanks for edits - I'll take a look first thing next week.

**Comment by kescobo** [comment]
@tszanalytics Thanks for the edits. I am largely satisfied with the changes. There are just a few outstanding questions/concerns.

1. I missed this on my first go-through, but it seems like the citation numbers are out of order (eg. Citation 3 comes first, followed by citation 8 etc). I haven't used latex that much, but I thought that citations were put in order automatically. Is something interfering with that? If this is not expected/accepted behavior, I think the citations should be in order.
1. @matbesancon suggested that we should be evaluating both BayesTesting.jl and the content of the manuscript, but @tszanalytics says otherwise. Can we get clarification on that? I haven't thoroughly evaluated the package, and it should be made clear whether that's part of the review process.
1. I am happy to see the notebooks included for analysis. Can you please provide a `Project.toml` and `Manifest.toml` for (just one that includes all the packages necessary for both should be fine). This will be useful for future readers so they can get the exact versions of the packages you use. Also would be good to include the julia version number you're using I think.

**Comment by matbesancon** [comment]
For the second point, if the manuscript is about and uses BayesTesting.jl, it should be reviewed with it. Otherwise if the software is contained in the repo of the paper, it should be as suggested versioned & documented

**Comment by tszanalytics** [comment]
1.  I'm not sure about the citations.  It appears to put them in alphabetical order by first author name.  I am using 
\bibliographystyle{juliacon}
\bibliography{ref}
as the bibliography style as per instructions, so that must be what is wanted for the proceedings?

2.  My presentation and this paper is not about the package and only uses the package somewhat tangentially.  I presented a preliminary version of the package at JuliaCon 2018, and plan on updating and registering it to provide a more complete package by next year, so I don't think the package itself is ready for evaluation.  The paper and presentation is about application of Julia to analysis of RCTs and meta analysis.

3.  Unfortunately I must admit that I don't really understand toml files and there usage as yet.  I will have to read up on this.  I believe I would have to specify a particular project to enable generation of toml files that contain only the packages for that project?  I'm not sure how to do this yet.  
I included the Julia version and computer specifics in the paper on page 3 (at end of right column). 

Thanks!

**Comment by bkamins** [comment]
Ad. 3 - this is very easy:

1. start a terminal in the directory of your project and run a Julia session
2. switch to package manager mode
3. write `activate .`
4. wtire `add WHATEVER_PACKAGES_YOU_REQUIRE_FOR_THE_CODES_TO_WORK`
5. exit Julia

In the directory of your project you will have the files @kescobo mentions.

**Comment by kescobo** [comment]
1. I agree. If this is the behavior of the styles provided, leave as is
1. Once (3) is done, Someone should be able to clone this repository, instantiate it, and then run the notebooks. When that's ready, let me know and I'll give it a shot.
1. Follow Bogumił's instructions and you should be good

**Comment by matbesancon** [comment]
thanks both for the review @tszanalytics could you confirm us when the changes are made?

**Comment by tszanalytics** [comment]
Yes!  Apologies for the delay.  I have been at a conference last week and
headed right back out to another one this week (currently in Chicago) -
everything happens at once!  I should be able to get to this within the
next few days.  Thanks to @bkamins for the clear guidance - it looks very
straightforward with that help.  I will upload the toml files asap.
Thanks for you patience.

Best,
Jeff

**Comment by tszanalytics** [comment]
@matbesancon 

Ok, the TOML files and updated versions of the notebooks for Julia 1.2.0 are uploaded, so it all should be in order now.

Again, apologies for the long delay.  I made the mistake of upgrading to julia 1.2 which created some problems with Turing, but everything is sorted now.

Thanks for your patience! 
Jeff

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00036/jcon.00036/10.21105.jcon.00036.pdf)

**Comment by matbesancon** [comment]
Ping @bkamins @kescobo

**Comment by bkamins** [comment]
I have updated the ticks in the original comment of this topic. I think the one minor thing that could be added (it is not blocking) is that README.md in https://github.com/tszanalytics/Juliacon2019 could have a bit more explanation what files constitute what part of the repository and possibly how they are intended to be used (as it contains: presentation, paper, notebooks and Julia files).

**Comment by matbesancon** [comment]
Ping @kescobo, there are a few boxes left

**Comment by kescobo** [comment]
Thanks for the ping. Just cloned and instantiated the repo - looks great! Boxes checked, thanks for your efforts @tszanalytics, and @matbesancon / @bkamins for helping with review :-) 👍

**Comment by matbesancon** [comment]
awesome, thanks both for the reviews :)

**Comment by matbesancon** [comment]
@whedon check references

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1089/cap.2016.0198 is OK
- 10.1016/j.jaac.2018.01.015 is OK
- 10.1089/cap.2017.0060 is OK
- 10.1089/cap.2018.0145 is OK
- 10.1080/01621459.2017.1316279 is OK
- 10.1016/j.csda.2010.01.019 is OK
- 10.1080/00031305.2019.1583913 is OK

MISSING DOIs

- https://doi.org/10.1137/141000671 may be missing for title: Julia: A fresh approach to numerical computing
- https://doi.org/10.1016/j.jaac.2019.10.013 may be missing for title: Antidepressant Tolerability in Pediatric Anxiety and Obsessive Compulsive Disorders: A Bayesian Hierarchical Modeling Meta-Analysis

INVALID DOIs

- 0.13140/RG.2.2.18951.70564 is INVALID
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00036/jcon.00036/10.21105.jcon.00036.pdf)

**Comment by matbesancon** [comment]
@tszanalytics sorry for the huge delay we had on the proceedings. A minor last thing before accepting, could you add the missing DOI pointed above to the references?

**Comment by matbesancon** [comment]
bump :)

**Comment by matbesancon** [comment]
Also as ultimate step, can you archive the project on Zenodo and let us know the DOI?

**Comment by matbesancon** [comment]
bump @tszanalytics

**Comment by vchuravy** [comment]
@tszanalytics bump

**Comment by vchuravy** [comment]
@whedon remind @tszanalytics in one week

**Comment by whedon** [comment]
Reminder set for @tszanalytics in one week

**Comment by whedon** [comment]
:wave: @tszanalytics, please update us on how things are progressing here (this is an automated reminder).

**Comment by kescobo** [comment]
Looks like there hasn't been any activity on that account since October of last year, and before that, February. Maybe worth reaching out by email (tszanalytics@gmail.com is what's associated with the account) or through other channels? I found his faculty profile pretty easily but not sure it's kosher to post that here.

**Comment by matbesancon** [comment]
Indeed no, thanks I'll reach out directly

**Comment by matbesancon** [comment]
Closing this for now since no response from the author. We'll gladly reopen when they contact us. Thanks to the reviewers still for their work on that!

**Comment by lucaferranti** [comment]
@editorialbot reject
